src/main/java/org/jenkinsci/plugins/github/webhook/WebhookManager.java

```diff
@@ -141,8 +141,22 @@ public void run() {
      */
     public void unregisterFor(GitHubRepositoryName name, List<GitHubRepositoryName> aliveRepos) {
         try {
-            GHRepository repo = checkNotNull(
-                    from(name.resolve(allowedToManageHooks())).firstMatch(withAdminAccess()).orNull(),
+            /* This is overcomplicated to satisfy unit tests that expect certain codepaths to be called */
+            GHRepository repo = from(name.resolve(allowedToManageHooks())).firstMatch(withAdminAccess()).orNull();
+
+            if (!(name.resolve(allowedToManageHooks()).iterator().hasNext())) {
```

Comment:
double call to name resolve doesn't look effective

Reply:
Good point, fixed.